### PR TITLE
Autonity contract e2e autotest framework

### PR DESCRIPTION
**Pull request by Jason-Zhangxin-Chen**
Autonity contract e2e test framework.

1. Python based e2e test for autontiy contract.
2. Test cases for autonity contract upgrade (include rollback of wrong contract bin&abi) and validator verification.
3. The test is integrated with CI in this submit.

### CODE REVIEW

**Comment by Jason-Zhangxin-Chen** [comment]
Chain parameter related to this contract upgrade machanism are recorded at wiki: 
https://github.com/clearmatics/autonity-wiki/wiki/Autonity-Chain-Parameter-Highlights

**Comment by Jason-Zhangxin-Chen** [comment]
> Why do we want use python for this purposes? Maybe this tests should be in separate repo?

Hi Eugene, the purpose was to test autontiy contract from e2e point of view, taking the client as a blackbox. On each changes of autontiy contract, they should be tested from e2e point of view. I thought we can integrate this at current repo since it is not a big stuff -- only one script.

File: params/autonity_contract_default.go [hunk 1]
```diff
@@ -5,1089 +5,6 @@ import "github.com/clearmatics/autonity/common"
...
...630781846159b6565b905092915050565b600060208201905081810360008301526163298184615a2b565b905092915050565b60006020820190506163466000830184615b6b565b92915050565b600060208201905081810360008301526163668184615c4a565b905092915050565b600060208201905081810360008301526163888184615ba7565b905092915050565b600060408201905081810360008301526163aa8185615ba7565b905081810360208301526163be8184615ba7565b90509392505050565b600060208201905081810360008301526163e081615c83565b9050919050565b6000602082019050818103600083015261640081615cc3565b9050919050565b6000602082019050818103600083015261642081615d03565b9050919050565b6000602082019050818103600083015261644081615d43565b9050919050565b6000602082019050818103600083015261646081615d83565b9050919050565b6000602082019050818103600083015261648081615dc3565b9050919050565b600060208201905081810360008301526164a081615e29565b9050919050565b600060208201905081810360008301526164c081615e8f565b9050919050565b600060208201905081810360008301526164e081615ecf565b9050919050565b6000602082019050818103600083015261650081615f0f565b9050919050565b600060208201905081810360008301526165218184615f4f565b905092915050565b600060208201905081810360008301526165438184615fed565b905092915050565b6000602082019050616560
```
Comment (b00ris) [inline params/autonity_contract_default.go]: I think it's not a good idea to put Default ABI in one line. It's useful to see what arguments and return types you need for EVM

**Comment by b00ris** [comment]
@Jason-Zhangxin-Chen I think it's better to keep python script there - https://github.com/clearmatics/ATAutonity. And integrate it to autonity CI.

**Comment by Jason-Zhangxin-Chen** [comment]
@b00ris @JekaMas @yazzaoui, the scripts of the autonity contract e2e test was moved into seperate repo: https://github.com/clearmatics/ATAutonity/tree/master/contract-e2e-test.
Please review and approve current PR since there was minor changes on the autonity contract side for the contract versioning.

**Pull request merged**